cisco_aci/datadog_checks/cisco_aci/api.py

```diff
@@ -60,11 +102,51 @@ def __init__(self, aci_urls, username, password, verify=False, timeout=10, log=N
         # This is used in testing
         self._refresh_sessions = True
 
+        self.password = password
+
+        self.cert_key_password = cert_key_password
+
+        self.cert_name = None
+        self.cert_key = None
+        if cert_name and cert_key:
+            self.cert_name = cert_name
+            self.cert_key = cert_key
+        elif not password:
+            msg = "You need to have either a password or a cert"
+            raise ConfigurationException(msg)
+
     def close(self):
         for session in self.sessions:
             session.close()
 
-    def login(self):
+    def setup_cert_login(self):
+        if self._refresh_sessions:
+            # ensure sessions are an empty array
+            self.sessions = []
+        for aci_url in self.aci_urls:
```

Comment:
It seems to me we aren't actually breaking out of this loop when a session is found or created. Is this desired?

Reply:
Hmm, I don't quite understand. Can you please explain?

Reply:
we want to create a session for each

Reply:
This is a setup function, it happens before any sessions are used. The actual requests are made in another part of the API.

Reply:
🤦‍♂️ I was seeing the 2nd if block as outside the for loop...

Reply:
This is why python needs brackets or an end operator